### PR TITLE
Feature: Download CSV

### PR DESCRIPTION
**Pull request by dustinlarimer**
## What does this PR do?

This PR builds off of https://github.com/keen/explorer/pull/190, a PR created by @menismu, which offers users the ability to download query response as a CSV file. This feature was requested in https://github.com/keen/explorer/issues/94. Thanks @menismu for implementing this frequently requested feature!

**Updates in this PR:**
* Sets filename as "untitled-query.csv" when a title has not been set
* Changes the button text to "Download CSV"
* Fixes margin/spacing of the download button

**Screenshot:**

screen shot 2016-12-20 at 2 18 46 pm

## How should this be tested?

- Get the branch running locally and run the tests with `npm run test`.
- QA the download behavior in the browser. Test several user flows (e.g. download an unnamed query, download a named query)

## Related Tickets
https://github.com/keen/explorer/pull/190, #94

### CODE REVIEW

File: client/js/app/components/explorer/visualization/index.js [hunk 1]
```diff
@@ -88,6 +89,12 @@ var Visualization = React.createClass({
     AppDispatcher.unregister(this.dispatcherToken);
   },
 
+  exportToCsv: function() {
```
Comment (sfalkoff) [inline client/js/app/components/explorer/visualization/index.js]: nit: two spaces instead of tabs

File: client/js/app/utils/DataUtils.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+var _ = require('lodash');
+
+module.exports = {
+  exportToCsv: function(data, filename) {
+	  var csvContent = "data:text/csv;charset=utf-8,";
```
Comment (sfalkoff) [inline client/js/app/utils/DataUtils.js]: nit: two spaces instead of tabs

File: client/js/app/components/explorer/visualization/index.js (hunk 1 above)
Comment (dustinlarimer) [inline client/js/app/components/explorer/visualization/index.js]: I'm not seeing this in the code.. what is displaying on your end?

File: client/js/app/utils/DataUtils.js (hunk 2 above)
Comment (dustinlarimer) [inline client/js/app/utils/DataUtils.js]: Nice catch- also lodash doesn't appear to be used here. Removing..

File: client/js/app/components/explorer/visualization/index.js (hunk 1 above)
Comment (dustinlarimer) [inline client/js/app/components/explorer/visualization/index.js]: I think this has been resolved in my latest commit.. or maybe I'm not seeing the same thing.

File: test/unit/components/explorer/visualization/keen_viz_spec.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  beforeEach(function() {
+    this.model = TestHelpers.createExplorerModel();
+    this.dataviz = TestHelpers.createDataviz();
+    this.exportToCsv = function() { return this; }
+    this.component = TestUtils.renderIntoDocument(<KeenViz model={this.model} dataviz={this.dataviz} exportToCsv={this.exportToCsv}/>);
+    this.exportToCsvStub = sinon.stub(this, 'exportToCsv');
+  });
+
+  describe('export to csv button', function() {
+    it('expor to csv button is not shown when chart type is different than table', function() {
```
Comment (sfalkoff) [inline test/unit/components/explorer/visualization/keen_viz_spec.js]: nit: typo "export"

**Comment by sfalkoff** [comment]
The padding around the Download CSV button looks a little different for me locally than pictured in the PR description -- does this look as expected? 

screen shot 2016-12-20 at 3 46 48 pm

**Comment by dustinlarimer** [comment]
@sfalkoff fixed in my latest PR- that was an edge-case for the extraction view. Nice catch!

File: client/js/app/components/explorer/visualization/index.js (hunk 1 above)
Comment (sfalkoff) [inline client/js/app/components/explorer/visualization/index.js]: Thanks for updating.

File: test/unit/components/explorer/visualization/keen_viz_spec.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    this.exportToCsv = function() { return this; }
+    this.component = TestUtils.renderIntoDocument(<KeenViz model={this.model} dataviz={this.dataviz} exportToCsv={this.exportToCsv}/>);
+    this.exportToCsvStub = sinon.stub(this, 'exportToCsv');
+  });
+
+  describe('export to csv button', function() {
+    it('export to csv button is not shown when chart type is different than table', function() {
+    	assert.lengthOf($R(this.component).find('[role="export-table"]').components, 0);
+    });
+    it('export to csv button is shown when char type table is selected', function() {
```
Comment (sfalkoff) [inline test/unit/components/explorer/visualization/keen_viz_spec.js]: nit: typo "char"

**Comment by sfalkoff** [comment]
Outside of the case where props or their values contain commas, this seems to be working great!

File: client/js/app/utils/DataUtils.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+module.exports = {
+  exportToCsv: function(data, filename) {
+    var csvContent = 'data:text/csv;charset=utf-8,';
+
+    data.forEach(function(infoArray, index){
+      dataString = infoArray.join(',');
+      csvContent += index < data.length ? dataString+ '\n' : dataString;
```
Comment (sfalkoff) [inline client/js/app/utils/DataUtils.js]: nit: missing space `dataString+`

**Pull request merged**